### PR TITLE
Upgrade ruma

### PR DESCRIPTION
**Pull request by aledomu**

### CODE REVIEW

**Review by aledomu — COMMENTED** [review]
Nevermind, just added another error variant.

File: matrix_sdk/src/client.rs [hunk 1]
```diff
@@ -727,11 +727,13 @@ impl Client {
...
     /// ```no_run
     /// use matrix_sdk::Client;
+    /// # use matrix_sdk_common::identifiers::ServerName;
+    /// # use std::convert::TryFrom;
     /// # use url::Url;
     /// # let homeserver = Url::parse("http://example.com").unwrap();
     /// # let limit = Some(10);
     /// # let since = Some("since token");
-    /// # let server = Some("server name");
+    /// # let server = Some(<&ServerName>::try_from("servername.com").unwrap());
```
Comment (jplatte) [inline matrix_sdk/src/client.rs:736]: `"servername.com".try_into().unwrap()` should work I think (looks a bit less exotic :slightly_smiling_face:)

**Review by poljar — CHANGES_REQUESTED** [review]
Please bump the type length limit in the WASM example over [here](https://github.com/matrix-org/matrix-rust-sdk/blob/master/matrix_sdk/examples/wasm_command_bot/src/lib.rs#L1) so the CI doesn't complain anymore:
```rust
#![type_length_limit="1702124"]
```

The rest looks good, thanks.

**Comment by aledomu** [comment]
It should be good now

**Pull request merged**